data-transfer-service/src/main/java/com/epam/pipeline/dts/transfer/service/impl/DtsSynchronizationService.java

```diff
@@ -0,0 +1,113 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.dts.transfer.service.impl;
+
+import com.epam.pipeline.dts.common.service.CloudPipelineAPIClient;
+import com.epam.pipeline.dts.transfer.model.pipeline.PipelineCredentials;
+import com.epam.pipeline.entity.dts.submission.DtsRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+@Service
+@Slf4j
+@EnableScheduling
+public class DtsSynchronizationService {
+
+    private final ConcurrentMap<String, String> preferences = new ConcurrentHashMap<>();
+    private final PipelineCredentials credentials;
+    private final ConfigurableApplicationContext context;
+    private final String dtsName;
+    private final String dtsLocalShutdownKey;
+
+    @Autowired
+    public DtsSynchronizationService(final @Value("${dts.local.api.url}") String apiUrl,
+                                     final @Value("${dts.local.name}") String dtsName,
+                                     final @Value("${dts.local.api.token}") String apiToken,
+                                     final @Value("${dts.local.preference.shutdown.key:dts.local.restart}")
+                                             String dtsLocalShutdownKey,
+                                     final ConfigurableApplicationContext context) {
+        this.credentials = new PipelineCredentials(apiUrl, apiToken);
+        this.dtsLocalShutdownKey = dtsLocalShutdownKey;
+        this.context = context;
+        this.dtsName = tryBuildDtsName(dtsName);
+        log.info("DTS sync is enabled for current host: `{}`.", this.dtsName);
+    }
+
+    @Scheduled(fixedDelayString = "${dts.local.preferences.poll:60000}")
+    public void synchronizePreferences() {
+        final Map<String, String> updatedPreferences = getApiClient().tryLoadDtsRegistryByNameOrId(dtsName)
+            .map(DtsRegistry::getPreferences)
+            .orElse(Collections.emptyMap());
+        if (shutdownRequired(updatedPreferences)) {
+            performShutdown();
+        }
+        preferences.clear();
+        preferences.putAll(updatedPreferences);
+    }
+
+    private CloudPipelineAPIClient getApiClient() {
+        return CloudPipelineAPIClient.from(credentials);
```

Comment:
Let's define `CloudPipelineAPIClient` bean in `com.epam.pipeline.dts.configuration.CommonConfiguration`.

Reply:
Done